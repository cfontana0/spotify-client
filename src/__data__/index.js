export const profile = {
  country: 'ES',
  display_name: 'Carlos',
  email: 'cfontana0@gmail.com',
  external_urls: {
    spotify: 'https://open.spotify.com/user/2qpe7xo6uc424zm2s64gyppwa',
  },
  followers: { href: null, total: 0 },
  href: 'https://api.spotify.com/v1/users/2qpe7xo6uc424zm2s64gyppwa',
  id: '2qpe7xo6uc424zm2s64gyppwa',
  images: [
    {
      height: null,
      url:
        'https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=10218601008714709&height=200&width=200&ext=1557472835&hash=AeR501syOinLmuJu',
      width: null,
    },
  ],
  product: 'open',
  type: 'user',
  uri: 'spotify:user:2qpe7xo6uc424zm2s64gyppwa',
};

export const playlist = {
  href:
    'https://api.spotify.com/v1/users/2qpe7xo6uc424zm2s64gyppwa/playlists?offset=0&limit=20',
  items: [
    {
      collaborative: false,
      external_urls: {
        spotify: 'https://open.spotify.com/playlist/4UKDDT9CtGcjKtVD9RqUDV',
      },
      href: 'https://api.spotify.com/v1/playlists/4UKDDT9CtGcjKtVD9RqUDV',
      id: '4UKDDT9CtGcjKtVD9RqUDV',
      images: [
        {
          height: 640,
          url:
            'https://mosaic.scdn.co/640/0ac84bcaa5a87297423e0d8ea0519f099d931477879256601c599ea61e823e2cd1a3c88f09ec941aea141e23f9cac9e6b0178c4e2f6cb61c347157a7ec5870eec53b65c861c40fefb443284dc91202c8',
          width: 640,
        },
        {
          height: 300,
          url:
            'https://mosaic.scdn.co/300/0ac84bcaa5a87297423e0d8ea0519f099d931477879256601c599ea61e823e2cd1a3c88f09ec941aea141e23f9cac9e6b0178c4e2f6cb61c347157a7ec5870eec53b65c861c40fefb443284dc91202c8',
          width: 300,
        },
        {
          height: 60,
          url:
            'https://mosaic.scdn.co/60/0ac84bcaa5a87297423e0d8ea0519f099d931477879256601c599ea61e823e2cd1a3c88f09ec941aea141e23f9cac9e6b0178c4e2f6cb61c347157a7ec5870eec53b65c861c40fefb443284dc91202c8',
          width: 60,
        },
      ],
      name: 'Acoustic',
      owner: {
        display_name: 'Basilio',
        external_urls: {
          spotify: 'https://open.spotify.com/user/2qpe7xo6uc424zm2s64gyppwa',
        },
        href: 'https://api.spotify.com/v1/users/2qpe7xo6uc424zm2s64gyppwa',
        id: '2qpe7xo6uc424zm2s64gyppwa',
        type: 'user',
        uri: 'spotify:user:2qpe7xo6uc424zm2s64gyppwa',
      },
      primary_color: null,
      public: true,
      snapshot_id:
        'MTksMDJkNTc0NDAyYjQ5NDMyMjIzMDg1OGExZjgwOWRhNzNjMjFhZTNmZA==',
      tracks: {
        href:
          'https://api.spotify.com/v1/playlists/4UKDDT9CtGcjKtVD9RqUDV/tracks',
        total: 10,
      },
      type: 'playlist',
      uri: 'spotify:playlist:4UKDDT9CtGcjKtVD9RqUDV',
    },
    {
      collaborative: false,
      external_urls: {
        spotify: 'https://open.spotify.com/playlist/5Mg504qjerC46zDEttHbOt',
      },
      href: 'https://api.spotify.com/v1/playlists/5Mg504qjerC46zDEttHbOt',
      id: '5Mg504qjerC46zDEttHbOt',
      images: [
        {
          height: 640,
          url:
            'https://mosaic.scdn.co/640/17e7c4d56fce74365a19e5ad9bd280633eade7852caa8d2fface94ee9d58f7005e54384a0fc9c17a52d83377fe9d7fc838deeb859499f3973adf903fb48aed6518256335ebb92ef79ce9d52ac6b2955b',
          width: 640,
        },
        {
          height: 300,
          url:
            'https://mosaic.scdn.co/300/17e7c4d56fce74365a19e5ad9bd280633eade7852caa8d2fface94ee9d58f7005e54384a0fc9c17a52d83377fe9d7fc838deeb859499f3973adf903fb48aed6518256335ebb92ef79ce9d52ac6b2955b',
          width: 300,
        },
        {
          height: 60,
          url:
            'https://mosaic.scdn.co/60/17e7c4d56fce74365a19e5ad9bd280633eade7852caa8d2fface94ee9d58f7005e54384a0fc9c17a52d83377fe9d7fc838deeb859499f3973adf903fb48aed6518256335ebb92ef79ce9d52ac6b2955b',
          width: 60,
        },
      ],
      name: 'Rock',
      owner: {
        display_name: 'Basilio',
        external_urls: {
          spotify: 'https://open.spotify.com/user/2qpe7xo6uc424zm2s64gyppwa',
        },
        href: 'https://api.spotify.com/v1/users/2qpe7xo6uc424zm2s64gyppwa',
        id: '2qpe7xo6uc424zm2s64gyppwa',
        type: 'user',
        uri: 'spotify:user:2qpe7xo6uc424zm2s64gyppwa',
      },
      primary_color: null,
      public: true,
      snapshot_id:
        'MTMsYmNjZGJlYjIxMzQ3NDQ4Nzg4Y2U4YmQ5YTYzZmE0M2RlMjM3OTQ1Yw==',
      tracks: {
        href:
          'https://api.spotify.com/v1/playlists/5Mg504qjerC46zDEttHbOt/tracks',
        total: 4,
      },
      type: 'playlist',
      uri: 'spotify:playlist:5Mg504qjerC46zDEttHbOt',
    },
    {
      collaborative: false,
      external_urls: {
        spotify: 'https://open.spotify.com/playlist/1EZNHYCutMm6a4O4074Dcv',
      },
      href: 'https://api.spotify.com/v1/playlists/1EZNHYCutMm6a4O4074Dcv',
      id: '1EZNHYCutMm6a4O4074Dcv',
      images: [
        {
          height: 640,
          url:
            'https://mosaic.scdn.co/640/6218fc701d5c6e66c200f29b57fc4cd5f979313f70d8660d8b802835a280ce75799d30db262f7094a8bfdddcca4dcc4c86184593a966f9561ba44c95ec5870eec53b65c861c40fefb443284dc91202c8',
          width: 640,
        },
        {
          height: 300,
          url:
            'https://mosaic.scdn.co/300/6218fc701d5c6e66c200f29b57fc4cd5f979313f70d8660d8b802835a280ce75799d30db262f7094a8bfdddcca4dcc4c86184593a966f9561ba44c95ec5870eec53b65c861c40fefb443284dc91202c8',
          width: 300,
        },
        {
          height: 60,
          url:
            'https://mosaic.scdn.co/60/6218fc701d5c6e66c200f29b57fc4cd5f979313f70d8660d8b802835a280ce75799d30db262f7094a8bfdddcca4dcc4c86184593a966f9561ba44c95ec5870eec53b65c861c40fefb443284dc91202c8',
          width: 60,
        },
      ],
      name: 'Relax',
      owner: {
        display_name: 'Basilio',
        external_urls: {
          spotify: 'https://open.spotify.com/user/2qpe7xo6uc424zm2s64gyppwa',
        },
        href: 'https://api.spotify.com/v1/users/2qpe7xo6uc424zm2s64gyppwa',
        id: '2qpe7xo6uc424zm2s64gyppwa',
        type: 'user',
        uri: 'spotify:user:2qpe7xo6uc424zm2s64gyppwa',
      },
      primary_color: null,
      public: true,
      snapshot_id:
        'MTMsOGMzZjFmOTE0ZGUyYWY1NmExZmJhZDUxY2IwYzM4YmJkY2U3YWM0Mw==',
      tracks: {
        href:
          'https://api.spotify.com/v1/playlists/1EZNHYCutMm6a4O4074Dcv/tracks',
        total: 6,
      },
      type: 'playlist',
      uri: 'spotify:playlist:1EZNHYCutMm6a4O4074Dcv',
    },
  ],
  limit: 20,
  next: null,
  offset: 0,
  previous: null,
  total: 3,
};

export const songs = {
  items: [
    {
      added_at: '2019-04-02T07:11:55Z',
      added_by: {
        external_urls: {
          spotify: 'https://open.spotify.com/user/2qpe7xo6uc424zm2s64gyppwa',
        },
        href: 'https://api.spotify.com/v1/users/2qpe7xo6uc424zm2s64gyppwa',
        id: '2qpe7xo6uc424zm2s64gyppwa',
        type: 'user',
        uri: 'spotify:user:2qpe7xo6uc424zm2s64gyppwa',
      },
      is_local: false,
      primary_color: null,
      track: {
        album: {
          album_type: 'album',
          artists: [
            {
              external_urls: {
                spotify:
                  'https://open.spotify.com/artist/4gzpq5DPGxSnKTe4SA8HAU',
              },
              href: 'https://api.spotify.com/v1/artists/4gzpq5DPGxSnKTe4SA8HAU',
              id: '4gzpq5DPGxSnKTe4SA8HAU',
              name: 'Coldplay',
              type: 'artist',
              uri: 'spotify:artist:4gzpq5DPGxSnKTe4SA8HAU',
            },
          ],
          external_urls: {
            spotify: 'https://open.spotify.com/album/0RHX9XECH8IVI3LNgWDpmQ',
          },
          href: 'https://api.spotify.com/v1/albums/0RHX9XECH8IVI3LNgWDpmQ',
          id: '0RHX9XECH8IVI3LNgWDpmQ',
          images: [
            {
              height: 640,
              url:
                'https://i.scdn.co/image/70d8660d8b802835a280ce75799d30db262f7094',
              width: 640,
            },
            {
              height: 300,
              url:
                'https://i.scdn.co/image/671fb0a2ecb0c77cb693eb150bde7b6fa94b3f32',
              width: 300,
            },
            {
              height: 64,
              url:
                'https://i.scdn.co/image/fb5bf70413af17edae091532e9f0dc442af852cf',
              width: 64,
            },
          ],
          name: 'A Rush Of Blood To The Head',
          release_date: '2002-08-08',
          release_date_precision: 'day',
          total_tracks: 11,
          type: 'album',
          uri: 'spotify:album:0RHX9XECH8IVI3LNgWDpmQ',
        },
        artists: [
          {
            external_urls: {
              spotify: 'https://open.spotify.com/artist/4gzpq5DPGxSnKTe4SA8HAU',
            },
            href: 'https://api.spotify.com/v1/artists/4gzpq5DPGxSnKTe4SA8HAU',
            id: '4gzpq5DPGxSnKTe4SA8HAU',
            name: 'Coldplay',
            type: 'artist',
            uri: 'spotify:artist:4gzpq5DPGxSnKTe4SA8HAU',
          },
        ],
        available_markets: [
          'AD',
          'AE',
          'AR',
          'AT',
          'AU',
          'BE',
          'BG',
          'BH',
          'BO',
          'BR',
          'CA',
          'CH',
          'CL',
          'CO',
          'CR',
          'CY',
          'CZ',
          'DE',
          'DK',
          'DO',
          'DZ',
          'EC',
          'EE',
          'EG',
          'ES',
          'FI',
          'FR',
          'GB',
          'GR',
          'GT',
          'HK',
          'HN',
          'HU',
          'ID',
          'IE',
          'IL',
          'IN',
          'IS',
          'IT',
          'JO',
          'JP',
          'KW',
          'LB',
          'LI',
          'LT',
          'LU',
          'LV',
          'MA',
          'MC',
          'MT',
          'MX',
          'MY',
          'NI',
          'NL',
          'NO',
          'NZ',
          'OM',
          'PA',
          'PE',
          'PH',
          'PL',
          'PS',
          'PT',
          'PY',
          'QA',
          'RO',
          'SA',
          'SE',
          'SG',
          'SK',
          'SV',
          'TH',
          'TN',
          'TR',
          'TW',
          'US',
          'UY',
          'VN',
          'ZA',
        ],
        disc_number: 1,
        duration_ms: 309600,
        episode: false,
        explicit: false,
        external_ids: { isrc: 'GBAYE0200772' },
        external_urls: {
          spotify: 'https://open.spotify.com/track/75JFxkI2RXiU7L9VXzMkle',
        },
        href: 'https://api.spotify.com/v1/tracks/75JFxkI2RXiU7L9VXzMkle',
        id: '75JFxkI2RXiU7L9VXzMkle',
        is_local: false,
        name: 'The Scientist',
        popularity: 84,
        preview_url:
          'https://p.scdn.co/mp3-preview/95cb9df1b056d759920b5e85ad7f9aff0a390671?cid=860725a4e08640199e5261d2a8a5d07c',
        track: true,
        track_number: 4,
        type: 'track',
        uri: 'spotify:track:75JFxkI2RXiU7L9VXzMkle',
      },
      video_thumbnail: { url: null },
    },
  ],
};

export const error = 'There was an error trying to get your profile';
