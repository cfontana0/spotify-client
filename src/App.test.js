import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { profile, playlist, songs, error } from './__data__';
import Home from './containers/home';

const mockStore = configureStore([thunk]);
const store = mockStore({
  home: {
    playlist,
    profile,
    songs,
    error,
    automatedTest: true,
  },
});

jest.mock('react-h5-audio-player', () => () => <div />);

it('renders without crashing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
