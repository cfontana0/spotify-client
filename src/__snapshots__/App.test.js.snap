// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="ui fluid container home"
>
  <div
    className="ui inverted segment"
  >
    <div
      className="ui feed"
    >
      <div
        className="event menu-user"
      >
        <div
          className="label"
        >
          <img
            alt="Carlos"
            src="https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=10218601008714709&height=200&width=200&ext=1557472835&hash=AeR501syOinLmuJu"
          />
        </div>
        <div
          className="content"
        >
          <div
            className="summary"
          >
            <a
              className="user"
            >
              Carlos
            </a>
            <button
              className="ui circular icon button logout"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="log out icon"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h4
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="spotify icon"
    />
    <div
      className="content"
    >
      PLAYLIST:
       
      <div
        aria-expanded={false}
        className="ui inline dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          PICK ONE
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            className="header"
          >
            Adjust time span
          </div>
          <div
            aria-checked={true}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              PICK ONE
            </span>
          </div>
        </div>
      </div>
    </div>
  </h4>
  <div
    className="ui celled grid"
  >
    <div
      className="ten column row"
    >
      <div
        className="three wide column"
      >
        <p
          className="titleText"
        >
          PLAYLIST
        </p>
        <p
          onClick={[Function]}
        >
          Acoustic
        </p>
        <p
          onClick={[Function]}
        >
          Rock
        </p>
        <p
          onClick={[Function]}
        >
          Relax
        </p>
      </div>
      <div
        className="ui active transition visible dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui indeterminate text loader"
          >
            Loading Playlist
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui-alerts top-right "
  />
</div>
`;
